source/agora/flash/Channel.d

```diff
@@ -2233,7 +2233,14 @@ LOuter: while (1)
         auto update_input = update_tx.inputs[0];
         auto update_ouput = update_tx.outputs[0];
 
-        auto utxos = this.getFeeUTXOs(update_tx.sizeInBytes());
+        // calculate fee
+        auto max_size = update_tx.sizeInBytes();
+        auto mock_refund_output = Output(Amount.init, this.own_pk);
+        max_size += mock_refund_output.sizeInBytes();
+        auto mock_update_unlock = this.update_signer.makeUpdateUnlock(SigPair.init, update.seq_id);
+        max_size += mock_update_unlock.sizeInBytes();
+        auto utxos = this.getFeeUTXOs(max_size);
```

Comment:
This is a chicken&egg problem. We can't know how many UTXOs `getFeeUTXOs` will return. So any calculations we do here is an approximation of the TX size. We have to check if our TX size is the same as the expected size after adding fees, or delegate that work to the getFeeUTXOs itself and assume that returned UTXOs are enough to cover the additional inputs and outputs.

Reply:
Right, I missed a calculation of the fee for outputs. In fact, the code in `getFeeUTXOs` calculates the fee for inputs. I will add it and check that the returned UTXOs are enough. Thanks.

Reply:
After looking into the code again, the output for the refund of the fee was also calculated as follows.
`
auto mock_refund_output = Output(Amount.init, this.own_pk);
`
So only thing to be needed in this part is to add checking that the returned UTXOs are enough.

Reply:
Updated.